nb-apis/adapters-api/src/test/java/io/nosqlbench/adapters/api/templating/StrInterpolatorTest.java

```diff
@@ -115,6 +117,30 @@ public void shouldMatchWithComments() {
         assertThat(a).isEqualTo("START\n# TEMPLATE(blahblah,blah)\nUniform(0,1000000000)->int;");
     }
 
+    @Test
+    public void sustitutionTests() {
+        StrInterpolator interp = new StrInterpolator(abcd);
+        String a = interp.apply("-${setkey:=setme}-${setkey}-");
+        assertThat(a).isEqualTo("-setme-setme-");
+        String b = interp.apply("-${setkey2:-setme}-${setkey2:-setyou}-");
+        assertThat(b).isEqualTo("-setme-setyou-");
+        String c = interp.apply("-${setkey:=setme}-${setkey3:+setme}-${setkey:+setyou}-");
+        assertThat(c).isEqualTo("-setme-setme-setyou-");
+    }
+
```

Comment:
Can we add a check for the nested case of at least one level?

Reply:
Sure we can add any test you want.